src/hotspot/share/utilities/debug.cpp

```diff
@@ -645,10 +646,11 @@ void help() {
 extern "C" DEBUGEXPORT void pns(void* sp, void* fp, void* pc) { // print native stack
   Command c("pns");
   static char buf[O_BUFLEN];
-  Thread* t = Thread::current_or_null();
   // Call generic frame constructor (certain arguments may be ignored)
   frame fr(sp, fp, pc);
-  VMError::print_native_stack(tty, fr, t, false, -1, buf, sizeof(buf));
+  NativeStackPrinter nsp(Thread::current_or_null());
+  nsp.print_stack_from_frame(tty, fr, buf, sizeof(buf),
+                  false /* print_source_info */, -1 /* max stack */);
```

Comment:
Should this be aligned with tty or is the current formatting preferred? Alternatively it seems like this could all fit on one line, but it's up to you

Reply:
Fixed - thanks.